src/async_client.rs

```diff
@@ -790,3 +798,88 @@ impl AsyncClient {
         Ok(response)
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::{AsyncClient, Url};
+    use crate::events::collections::all::RoomEvent;
+    use crate::identifiers::{RoomId, UserId};
+
+    use crate::test_builder::EventBuilder;
+    use crate::{assert_eq_, async_assert};
+
+    use std::convert::TryFrom;
+
+    #[tokio::test]
+    async fn client_runner() {
+        // TODO make this actually test something
+
+        async_assert! {
```

Comment:
Doing tests like this seems a bit complex, what's wrong with directly checking this assert instead of passing it to the client runner?


Helpers to build the events are a great idea though. If some part ends up being boilerplaty we can add macros for those parts.

Reply:
Oh yea that would be much simpler good thinking! Just have the `TestRunner` return the respective thing ie. `ClientTestRunner` returns the client, `MockTestRunner` returns the client and RoomTestRunner returns the Room? Then the tests can just assert right there no need for all the macro stuff I wrote.

Reply:
Yeah, something like that.